### PR TITLE
🐐 Bring more inline with gulp best practices

### LINKED ISSUE #289: Gulp tasks should be named functions wrapped and without runSequence
The gulp tasks have been written against gulp convention.  The true intention of the gulp design is to write regular Javascript functions, and only make the task for calling them. With this format, you would never need `runSequence` which uses(or used to use) `gulp.start()`. 

If you need sync, use a promise library like [blue bird performance promise library](https://github.com/petkaantonov/bluebird/blob/master/API.md). I believe Gulp 4.0 has it's own promise's but I prefer performance bluebird. 

This is from the authors of Gulp. Sources:
[gulpjs/gulp#755](https://github.com/gulpjs/gulp/issues/755#issuecomment-61884019)
[gulpjs/gulp#458](https://github.com/gulpjs/gulp/issues/458#issuecomment-62057370)
example on proper convention:

```javascript
function getJsFiles() {
    var sourcePaths = [
        './app/scripts/**/*.js',
        '!./app/scripts/**/*.spec.js',
        '!./app/scripts/app.js'
    ];

    var sources = gulp.src(sourcePaths, { read: false }).pipe(angularFilesort());

    return gulp.src('./app/index.html')
        .pipe(injector(sources, { ignorePath: 'app', addRootSlash: false }))
        .pipe(gulp.dest('./app'));
} 
```
----------------------------
```javascript 
gulp.task('js', function () {
    jsStream = getJsFiles();
});
```

### PR DESCRIPTION
**Pull request by samccone**
Extract non-specific tasks to fns to be more inline with "the gulp way✨"

Fixes #289

### CODE REVIEW

**Comment by addyosmani** [comment]
Reviewed.

**Comment by samccone** [comment]
updated!

File: gulpfile.js [hunk 1]
```diff
@@ -49,6 +49,50 @@ var styleTask = function (stylesPath, srcs) {
     .pipe($.size({title: stylesPath}));
 };
 
+var jshintTask = function (src) {
+  return gulp.src(src)
+    .pipe(reload({stream: true, once: true}))
```
Comment (didrocks) [inline gulpfile.js]: Hey, it seems to me that you are reintroducing the double reload that was removed as per PR #370, is that intended?

Comment (samccone) [inline gulpfile.js]: I just need to rebase and it should go away (unless Git gets really confused)👌 good spot though I will doubly make sure it gets removed!

**Comment by samccone** [comment]
yep it went away @didrocks :)

**Comment by didrocks** [comment]
@samccone great! thanks for confirming ;)

**Comment by addyosmani** [comment]
Tried running locally again. Looks like we're now breaking on the HTML task and are running into some linting issues:

```
polymer-starter-kit/gulpfile.js
  line 58  col 2   Missing semicolon.
  line 68  col 2   Missing semicolon.
  line 93  col 2   Missing semicolon.
  line 91  col 21  'dest' is not defined.

  ⚠  4 warnings
```

**Comment by samccone** [comment]
should be GTG now :checkered_flag:

**Comment by addyosmani** [comment]
Tested again and everything is passing now. Have some :cake:!

**Pull request closed without merging**